wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java

```diff
@@ -96,7 +96,8 @@ public static void setPropertyOnTarget(
       } else {
         writeMethod.invoke(target, propValue);
       }
-      LOGGER.finest(() -> String.format("Set property '%s' with value: %s", propName, propValue));
+      Object loggedProperty = propValue.equals(PropertyDefinition.PASSWORD.name) ? "***" : propValue;
```

Comment:
I'd extract a code that check if the property contains a secret into a separate function and use it everywhere here in PropertyUtils